src/ophyd_async/core/flyer.py

```diff
@@ -71,17 +65,19 @@ def prepare(self, value: T) -> AsyncStatus:
         return AsyncStatus(self._prepare(value))
 
     async def _prepare(self, value: T) -> None:
-        self._trigger_info = self._trigger_logic.trigger_info(value)
         # Move to start and setup the flyscan
         await self._trigger_logic.prepare(value)
 
     @AsyncStatus.wrap
     async def kickoff(self) -> None:
-        self._fly_status = AsyncStatus(self._trigger_logic.start())
+        self._fly_status = AsyncStatus(self._trigger_logic.kickoff())
+        await self._fly_status
 
-    def complete(self) -> AsyncStatus:
+    @AsyncStatus.wrap
+    async def complete(self) -> None:
         assert self._fly_status, "Kickoff not run"
-        return self._fly_status
+        await self._fly_status
```

Comment:
I don't think we should bother checking that kickoff hasn't been done, at least not at the moment, it requires too much holding of state for too little benefit...
```suggestion
```

Reply:
Yes I agree